app/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/RecipientLayoutCreator.kt

```diff
@@ -17,10 +21,21 @@ private const val LIST_SEPARATOR = ", "
 internal class RecipientLayoutCreator(
     private val textMeasure: TextMeasure,
     private val maxNumberOfRecipientNames: Int,
-    private val recipientsFormat: String,
+    val recipientsFormat: String,
     private val additionalRecipientSpacing: Int,
     private val additionalRecipientsPrefix: String,
 ) {
+    private val recipientsPrefix: String
+    private val recipientsSuffix: String
+
+    init {
+        val placeholderIndex = recipientsFormat.indexOf("%s")
+        require(placeholderIndex != -1) { "recipientFormat must contain '%s'" }
+
+        recipientsPrefix = recipientsFormat.substring(0, placeholderIndex)
+        recipientsSuffix = recipientsFormat.substring(placeholderIndex + 2)
```

Comment:
This could be simplified to :

```suggestion
        require(recipientsFormat.contains("%s")) { "recipientFormat must contain '%s'" }
        recipientsPrefix = recipientsFormat.substringBefore("%s")
        recipientsSuffix = recipientsFormat.substringAfter("%s")
```